minutes/models.py

```diff
@@ -41,3 +42,121 @@ def get_date_month(self):
 
     def get_date_day(self):
         return self.date.strftime("%d").zfill(2)
+
+
+class Concern(models.Model):
+    """ A "Concern" is a sort of Organizational Unit
+        which can exist in a tree like structure """
+
+    #  Name of the "Concern", Example: PSF Board, Packaging Working Group,
+    #  or "Project"
+    name = models.CharField(max_length=128)
+    #  If applicable, a "parent" concern.
+    #  Example: Finance committee of PSF Board,
+    #  or "PIP UX Project 2020" under the Packaging Working Group
+    parent_concern = models.ForeignKey('self', null=True, blank=True, on_delete=models.CASCADE, related_name="children_concerns")
+
+    def __str__(self):
+        if self.parent_concern_id:
+            return f"{self.parent_concern} > {self.name}"
+        return self.name
+
+    class Meta:
+        verbose_name = "concern"
+        verbose_name_plural = "concerns"
+
+    @property
+    def concerned_parties(self):
+        return ConcernedParty.objects.filter(role__in=self.concernrole_set.all())
+
+
+class ConcernRole(models.Model):
+    """ A User's role within a concern, may be generic and come with
+        certain specific items that they are responsible for or allowed
+        to provide """
+    name = models.CharField(max_length=128)
+    concern = models.ForeignKey(Concern, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f"{self.concern}: {self.name}"
+
+    class Meta:
+        verbose_name = "concern role"
+        verbose_name_plural = "concern roles"
+
+
+class ConcernedPartyQuerySet(models.QuerySet):
+    def by_concern(self, concern):
+        return self.filter(role__concern__name__iexact=concern)
+
+    def by_concern_role(self, concern, role):
+        return self.by_concern(concern).filter(role__name__iexact=role)
+
+
+class ConcernedParty(models.Model):
+    """ Effectively a "Group" for a given concern that makes managing
+        access and reminders more straightforward. This may include the ability
+        to email a notification that a meeting has been scheduled, remind
+        attendees to submit reports, etc. """
+    objects = ConcernedPartyQuerySet.as_manager()
+
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    role = models.ForeignKey(ConcernRole, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f"{self.user} as {self.role}"
+
+    @property
+    def display_name(self):
+        return self.user.get_full_name() or self.user.username
+
+    class Meta:
+        verbose_name = "concerned party"
+        verbose_name_plural = "concerned parties"
+
+
+class Meeting(models.Model):
+    title = models.CharField(max_length=500)
+    date = models.DateField(db_index=True)
+    started_at = models.DateTimeField(null=True, blank=True)
+    parties = models.ManyToManyField(ConcernedParty)
```

Comment:
By default, I think we'd generally be relating "attendees" to the `Meeting` object by `ConcernedRole`. Individually specifying parties is tedious.

Supporting ad hoc attendees is a good idea, but this field should probably be renamed `additional_parties` or `ad_hoc_parties` and be hidden under the fold in Django admin.

Reply:
Got it @ewdurbin! I thought about this `parties` m2m as a way to decouple the `Meeting` model from the concerns/roles structure. My initial idea was that the UX to create a new meeting would only allow the user to select many `ConcernedRole`. Then, the backend would fetch for the parties from the roles list and add them to this attribute. I thought that we would be able to enable an easy way to create meetings, but still allow a granular control when editing that. But I'm ok with removing this and leaving a m2m field to `ConcernedRole` model.

About the `ad_hoc_parties` I'm a +1 for keeping it separated from the primary list. But, just to confirm, shouldn't this attribute be a m2m to the `User` model? By doing so, we won't required the user to create a `ConcernedParty` entry for that user who will join a single meeting.

Reply:
Aha! That makes total sense actually, decoupling in that way is a good idea since members of a given `ConcernedRole` will change over time. 👍

As far as relating `ad_hoc_parties`, I think it's best if we keep that related to a `ConcernedParty`. Again this circles back to the fact that there are a _multitude_ of `User` objects in the database.